tests/test_healthsystem.py

```diff
@@ -1906,14 +2036,246 @@ def apply(self, person_id, squeeze_factor):
+            [{"Facility_ID": 20.0, "Officer_Type_Code": "DCSA", "Clinic1": 0.6, "GenericClinic": 0.4}]
+        )
+        sim.modules["HealthSystem"].parameters["clinic_mapping"] = pd.DataFrame(
+            [{"Treatment": "DummyHSIEvent", "Clinic": "Clinic1"}]
+        )
+        sim.modules["HealthSystem"]._clinic_names = ["Clinic1", "GenericClinic"]
+        sim.modules["HealthSystem"].setup_daily_capabilities("funded_plus")
+
+        # Assign the entire population to the first district, so that all events are run in the same district
+        col = "district_of_residence"
+        s = sim.population.props[col]
+        ## Not specifying the dtype explicitly here made the col a string rather than a category
+        ## and that caused problems later on.
+        sim.population.props[col] = pd.Series(s.cat.categories[0], index=s.index, dtype=s.dtype)
+
+        sim.simulate(end_date=sim.start_date + pd.DateOffset(years=1))
+
+        return sim
+
+    def schedule_hsi_events(notherclinic, nclinic1, sim):
+        for i in range(0, notherclinic):
+            hsi = DummyHSIEvent(
+                module=sim.modules["DummyModuleGenericClinic"],
+                person_id=i,
+                appt_type="ConWithDCSA",
+                level="0",
+                treatment_id="DummyHSIEventGenericClinic",
+            )
+            sim.modules["HealthSystem"].schedule_hsi_event(
+                hsi, topen=sim.date, tclose=sim.date + pd.DateOffset(days=1), priority=1
+            )
+
+        for i in range(notherclinic, notherclinic + nclinic1):
+            hsi = DummyHSIEvent(
+                module=sim.modules["DummyModuleClinic1"],
+                person_id=i,
+                appt_type="ConWithDCSA",
+                level="0",
+                treatment_id="DummyHSIEvent",
+            )
+            sim.modules["HealthSystem"].schedule_hsi_event(
+                hsi, topen=sim.date, tclose=sim.date + pd.DateOffset(days=1), priority=1
+            )
+
+        return sim
+
+    tot_population = 100
+    sim = create_simulation(tmpdir, tot_population)
+
+    ## Test that capabilities are split according the proportion specified for the Facility Id
+    ## and officer combination in the Resource file.
+    ## 40% of capabilities are GenericClinic and 60% Clinic1 capabilities
+    other_clinic = sim.modules["HealthSystem"]._daily_capabilities["GenericClinic"]
+    clinic1 = sim.modules["HealthSystem"]._daily_capabilities["Clinic1"]
+
+    # 'FacilityID_20_Officer_DCSA' is coming from the resource file
+    ratio = clinic1["FacilityID_20_Officer_DCSA"] / other_clinic["FacilityID_20_Officer_DCSA"]
+    expect = 0.6 / 0.4
+    assert abs(ratio - expect) < 1e-7, "GenericClinic capabilities are not split correctly"
+
+    # Schedule an identical appointment for all individuals, assigning clinic as follows:
+    # half individuals have clinic_eligibility=GenericClinic and half clinic_eligibility=Hiv
```

Comment:
Hiv is noted as the eligibility for the clinics, but the options for clinic actually appear to be: "Clinic1", "GenericClinic"